lib/pushmetrics/pushmetrics.go

```diff
@@ -59,3 +59,45 @@ func Stop() {
 	cancelPushCtx()
 	wgDone.Wait()
 }
+
+// StopAndPush stops the periodic push of metrics and performs the final push attempt.
+// It is recommended using StopAndPush for short-living applications, like vmbackup or vmrestore.
+// See Stop.
+//
+// StopAndPush must be called after Init.
+func StopAndPush() {
+	// stop periodic push of metrics
+	Stop()
+
+	if len(*pushURL) == 0 {
+		return
+	}
+
+	startTime := time.Now()
+	logger.Infof("pushing metrics on shutdown to configured -pushmetrics.url")
+
+	wg := sync.WaitGroup{}
+	extraLabels := strings.Join(*pushExtraLabel, ",")
+	pushedSuccessfullyTo := len(*pushURL)
+	for _, pu := range *pushURL {
+		// push to all destinations in parallel to speed up shutdown
+		wg.Add(1)
+		go func(pushURL string) {
+			ctxLocal, cancel := context.WithTimeout(context.Background(), *pushInterval)
+			opts := &metrics.PushOptions{
+				ExtraLabels:        extraLabels,
+				Headers:            *pushHeader,
+				DisableCompression: *disableCompression,
+			}
+			if err := metrics.PushMetricsExt(ctxLocal, pushURL, appmetrics.WritePrometheusMetrics, opts); err != nil {
+				logger.Errorf("failed to push metrics to %q: %s", pushURL, err)
+				pushedSuccessfullyTo--
```

Comment:
I suggest to convert this variable into atomic. Currently, it may cause data race.